src/Text/LLVM/AST.hs

```diff
@@ -1647,10 +1647,14 @@ data DISubprogram' lab = DISubprogram
 
 type DISubprogram = DISubprogram' BlockLabel
 
-data DISubrange = DISubrange
-  { disrCount      :: Int64
-  , disrLowerBound :: Int64
-  } deriving (Data, Eq, Generic, Ord, Show, Typeable)
+data DISubrange' lab = DISubrange
+  { disrCount      :: Either Int64 (Maybe (ValMd' lab))
+  , disrLowerBound :: Either Int64 (Maybe (ValMd' lab))
+  , disrUpperBound :: Maybe (ValMd' lab)
+  , disrStride     :: Maybe (ValMd' lab)
```

Comment:
If you happen to know this information, it would be helpful to document which LLVM version(s) introduced these new fields.

Reply:
I don't at this point.  I'm basing this off the latest master code from here: https://github.com/llvm/llvm-project/blob/bbe8cd13335300958b04db5318c31ff52714f96f/llvm/lib/Bitcode/Reader/MetadataLoader.cpp#L1435-L1461. 

It *will* be important to know the versions for the pretty-printing aspect (which I discuss further below).  Right now I'd like to restore llvm-pretty-bc-parser/llvm-pretty to operational status (particularly vis-a-vis PR 228 in the former). Once we are operational for the current CI-verified llvm version range, I plan to widen the version range via CI to identify where these version-specific inflection points are for pretty-printing (and we now have the machinery in the CI to do this).

Reply:
I took a look at the `git blame` in LLVM:

* `discrCount` can be an arbitrary metadata node as of [LLVM 7](https://github.com/llvm/llvm-project/commit/fdf40917d946e3b36fd67afb2aac4063af9fc85d), which corresponds to `case 1` in the LLVM code.
* `discrLowerBound` can be an arbitrary metadata node as of [LLVM 11](https://github.com/llvm/llvm-project/commit/d20bf5a7258d4b6a7f017a81b125275dac1aa166), which corresponds to `case 2` in the LLVM code.
* `discrUpperBound` and `discrStride` were introduced in LLVM 11, in the same commit that changed `discrLowerBound`.